core/state/statedb.go

```diff
@@ -845,12 +812,10 @@ func (s *StateDB) Finalise(deleteEmptyObjects bool) {
 			delete(s.accountsOrigin, obj.address) // Clear out any previously updated account data (may be recreated via a resurrect)
 			delete(s.storagesOrigin, obj.address) // Clear out any previously updated storage data (may be recreated via a resurrect)
 		} else {
+			obj.created = false
```

Comment:
Maybe add a doc here, seems a bit "arbitrary". why this is happening? I guess for non-newly-created objects this is a noop and for newly created ones this finalizes?

Reply:
Was thinking: we should rename created into `newContract`. Makes it a bit more obvious why it is unset here: it stops being new at the end of the tx.

Reply:
I think this flag should be revoked in `object.finalised()`, as it's a very internal thing inside of the object